plugins/SitesManager/Controller.php

```diff
@@ -130,84 +132,61 @@ public function siteWithoutData()
         $this->checkSitePermission();
 
         return $this->renderTemplateAs('siteWithoutData', [
-            'emailBody'                                           => SitesManager::renderTrackingCodeEmail($this->idSite),
-            'siteWithoutDataStartTrackingTranslationKey'          => StaticContainer::get('SitesManager.SiteWithoutDataStartTrackingTranslation'),
-            'inviteUserLink'                                      => $this->getInviteUserLink()
+            'inviteUserLink' => $this->getInviteUserLink()
         ], $viewType = 'basic');
     }
 
-    public function siteWithoutDataTabs()
+    public function getTrackingMethodsForSite()
```

Comment:
I already tried to reduce the complexity of that method and it's usage. But there is still certain potential for improvements. Building a repository / sorter / factory pattern for that single purpose only felt a bit over engineered in the first moment, but in terms of test-ability this would have a lot benefits indeed.

Reply:
Yeah exactly, not for now like I mentioned, but at least a consideration